### PR TITLE
Fix nightly build failure + Address issue with the self-hosted runner

### LINKED ISSUE #19712: Nightly Dotty workflow of 2024-02-17 failed
See https://github.com/lampepfl/dotty/actions/runs/7938874929

### PR DESCRIPTION
**Pull request by hamzaremmal**
Closes #19712 

This PR introduce a workaround for actions/runner#2033. This workaround was already introduced in #19703 in some of the jobs. 

`actions/checkout` has an issue filled about this too: actions/checkout#1169

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (hamzaremmal) [inline .github/workflows/ci.yaml:74]: The `true` at the end made the CI pass when it should have failed (See https://github.com/lampepfl/dotty/actions/runs/7924243033/job/21635455423?pr=19703). Should we remove it ?

File: .github/workflows/ci.yaml
Comment (hamzaremmal) [inline .github/workflows/ci.yaml:72]: The `true` at the end made the CI pass when it should have failed (See https://github.com/lampepfl/dotty/actions/runs/7924243033/job/21635455423?pr=19703). Should we remove it ?

**Comment by hamzaremmal** [comment]
I've added `git config --global --add safe.directory /__w/dotty/dotty` before each call to a git command as a workaround for the mentioned issue above.

**Comment by hamzaremmal** [comment]
Also, I believe that the issue didn't manifest earlier as the workflow "resets" the repository, then runs the `actions/checkout` action which then delete the content of the previous reset. Why are we reseting in that case ? Can we just do the checkout step without reseting the repository ?

**Comment by smarter** [comment]
I think this was to workaround issues with the repository getting stuck in a bad state with submodules, but it might not be necessary anymore as actions/checkout should now be more robust.

File: .github/workflows/ci.yaml
Comment (smarter) [reply .github/workflows/ci.yaml:72]: Sure.

File: .github/workflows/ci.yaml
Comment (smarter) [inline .github/workflows/ci.yaml:73]: Can you add a comment pointing to https://github.com/actions/runner/issues/2033 ?

**Review by smarter — APPROVED** [review]

File: .github/workflows/ci.yaml
Comment (hamzaremmal) [reply .github/workflows/ci.yaml:72]: I'll do it in another PR, see if I can remove the reset too.

**Comment by hamzaremmal** [comment]
I'll wait for the CI to finish the run and check if we don't have any other silent failure before merging it

**Pull request merged**
